Guides configure-redis-caching

JC3P0 opened the pull request:
[Guides] Add a scaling guide for caching with Redis

Fixes #112

JC3P0 committed:
Guides configure-redis-caching

[Guides] Add a scaling guide for caching with Redis

Fixes #112

JC3P0 commented:
I just went back and looked at the backlog and it looks like you wanted this in `/docs/scaling/cache/redis` not in the how-to-guides? lol oops, I can fix this if so.

Also, not sure if this was the way you wanted to show it getting set up with `docker network connect cerb-demo_backend redis-cerb` I could not get it to work without doing that even though it was not specified in the documentation https://hub.docker.com/_/redis

jstanden commented:
@JC3P0 The easiest way is just using `docker run redis` on a local portal like 6379. Then inside of Cerb, use `host.docker.internal` as the hostname for your local machine from inside the container.

That way you can connect to containers like Redis/Valkey/Elasticache/MySQL without them being in the container's private network.

JC3P0 commented:
@jstanden using `host.docker.internal` was much easier haha. I updated and force pushed the commit.

I moved the file to `/docs/scaling/cache/redis` but still not sure where it should be displayed? It looks like there is a `/docs/setup/configure/cache/` page from there we could link to this redis page? unless you want a scaling section on its own within the Admin guide?

jstanden commented:
@JC3P0 The `/docs` are a bit funky. Their index isn't currently automatic because we organize the docs into chapters with a specific order.

That layout is set here:
https://github.com/cerb/cerb.github.io/blob/11.0/_data/docs.yaml

We have an unlisted `/docs/scaling` page there that would have a section on server-side caches (Redis, Valkey, memcached).

Right now this can just go in `/guides/scaling/`.

I agree with linking to it from the Setup->Configure->Cache docs.

jstanden reviewed (approved):
This looks great. 👍🏻  I can move it into the right guide category.

I did change the default Cerb Dockerfile to include redis/memcached PHP extensions going forward.

jstanden commented:
Merged manually.
